src/Games/NexusMods.Games.MountAndBlade2Bannerlord/Diagnostics/MountAndBlade2BannerlordDiagnosticEmitter.cs

```diff
@@ -0,0 +1,200 @@
+            ),
+
+            // Conflicting Load Order: Load Before and After
+            ModuleDependencyConflictLoadBeforeAndAfterIssue conflictLoad => Diagnostics.CreateLoadOrderConflict(
+                ModId: conflictLoad.Module.Id,
+                ModName: conflictLoad.Module.Name,
+                ConflictingId: conflictLoad.ConflictingModule.Id
+            ),
+
+            // Circular Dependency
+            ModuleDependencyConflictCircularIssue circularIssue => Diagnostics.CreateCircularDependency(
+                ModId: circularIssue.Module.Id,
+                ModName: circularIssue.Module.Name,
+                CircularDependencyId: circularIssue.CircularDependency.Id,
+                CircularDependencyName: circularIssue.CircularDependency.Name
+            ),
+
+            // Load Order Not Met: Must Load Before
+            ModuleDependencyNotLoadedBeforeIssue notLoadedBefore => Diagnostics.CreateModMustLoadBefore(
+                ModId: notLoadedBefore.Module.Id,
+                ModName: notLoadedBefore.Module.Name,
+                DependencyId: notLoadedBefore.Dependency.Id
+            ),
+
+            // Load Order Not Met: Must Load After
+            ModuleDependencyNotLoadedAfterIssue notLoadedAfter => Diagnostics.CreateModMustLoadAfter(
+                ModId: notLoadedAfter.Module.Id,
+                ModName: notLoadedAfter.Module.Name,
+                DependencyId: notLoadedAfter.Dependency.Id
+            ),
+
+            // Configuration Error: Missing ID
+            ModuleMissingIdIssue missingId => Diagnostics.CreateModBadConfigMissingId(
+                ModName: missingId.Module.Name,
+                ModNameNoSpaces: missingId.Module.Name.Replace(" ", "")
+            ),
+
+            // Configuration Error: Missing Name
+            ModuleMissingNameIssue missingName => Diagnostics.CreateModBadConfigMissingName(
+                ModId: missingName.Module.Id
+            ),
+
+            // Configuration Error: Null Dependency
+            ModuleDependencyNullIssue nullDependency => Diagnostics.CreateModBadConfigNullDependency(
+                ModId: nullDependency.Module.Id,
+                ModName: nullDependency.Module.Name
+            ),
+
+            // Configuration Error: Dependency Missing ID
+            ModuleDependencyMissingIdIssue missingDependencyId => Diagnostics.CreateModBadConfigDependencyMissingId(
+                ModId: missingDependencyId.Module.Id,
+                ModName: missingDependencyId.Module.Name
+            ),
+
+            // Base classes, ignore.
+            ModuleVersionMismatchRangeIssue moduleVersionMismatchRangeIssue => null,
+            ModuleVersionMismatchSpecificIssue moduleVersionMismatchSpecificIssue => null,
+            ModuleVersionMismatchIssue moduleVersionMismatchIssue => null,
+            
+            // A new variant that's unknown to our code.
+            _ => throw new ArgumentOutOfRangeException(nameof(issue), issue, null),
```

Comment:
Should maybe be Debug assert?

Reply:
I was hoping to avoid the default case altogether when I originally wrote this, but the compiler wasn't letting me.

Ideally not handling a possible type should be a compile time error here. Can debug assert this, but then nobody would know if something was added, because nobody would ever report an issue.